morpheus/_lib/tests/llm/test_llm_task_handler_runner.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../test_utils/common.hpp"  // IWYU pragma: associated
+
+#include "morpheus/llm/input_map.hpp"
+#include "morpheus/llm/llm_context.hpp"
+#include "morpheus/llm/llm_lambda_node.hpp"
+#include "morpheus/llm/llm_node.hpp"
+#include "morpheus/llm/llm_node_runner.hpp"
+#include "morpheus/llm/llm_task.hpp"
+#include "morpheus/llm/llm_task_handler.hpp"
+#include "morpheus/llm/llm_task_handler_runner.hpp"
+#include "morpheus/types.hpp"
+
+#include <gtest/gtest.h>
+#include <mrc/channel/forward.hpp>
+#include <mrc/coroutines/sync_wait.hpp>
+#include <mrc/coroutines/task.hpp>
+
+#include <coroutine>
+#include <memory>
+#include <stdexcept>
+#include <string>
+
+using namespace morpheus;
+using namespace morpheus::test;
+using namespace mrc;
+
+class TestTaskHandler : public llm::LLMTaskHandler
+{
+  public:
+    TestTaskHandler() = default;
+
+    TestTaskHandler(std::vector<std::string> input_names)
+    {
+        m_input_names = input_names;
+    }
+
+    ~TestTaskHandler() override = default;
+
+    std::vector<std::string> get_input_names() const override
+    {
+        return m_input_names;
+    }
+
+    Task<return_t> try_handle(std::shared_ptr<llm::LLMContext> context) override
+    {
+        std::vector<std::shared_ptr<ControlMessage>> results;
+
+        for (auto& name : m_input_names)
+        {
+            auto msg_config = nlohmann::json();
+            nlohmann::json task_properties;
+            task_properties          = {{"task_type", "dictionary"}, {"model_name", "test"}};
+            task_properties["input"] = name;
+            msg_config["tasks"]      = {{{"type", "template"}, {"properties", task_properties}}};
+
+            auto msg = std::make_shared<ControlMessage>(msg_config);
+
+            results.push_back(msg);
+        }
+
+        co_return results;
+    }
+
+  private:
+    std::vector<std::string> m_input_names;
+};
+
+TEST_CLASS(LLMTaskHandlerRunner);
+
+TEST_F(TestLLMTaskHandlerRunner, TryHandle)
```

Comment:
Question: what are we testing here? it appears the code paths we hit are all part of the test handler, which only exists to serve the test.

Reply:
Thank you. I'm not even testing `LLMTaskHandlerRunner` here. I'll remove and use the test class only to test the non-virtual function `input_names()`.

Reply:
code paths now go though tested class